crates/matrix-sdk-ui/src/timeline/controller/mod.rs

```diff
@@ -647,9 +645,9 @@ impl<P: RoomDataProvider> TimelineController<P> {
                 .await;
         }
 
-        if !events.is_empty() {
+        if !state.items.is_empty() || !events.is_empty() {
```

Comment:
The previous check made sense because we used `clear` before, but now we may want to replace a populated timeline with an empty one.

Reply:
Add this in a comment please.